### ISSUE TITLE
Wrong comment line/column location

### ISSUE THREAD

**Issue by davidbonnet** [cherow/cherow#235]
Line/column locations of comments are off. In particular, end locations sometimes appear before start locations.
 
Furthermore, documentation should specify whether lines and columns are one- or zero-based.

**Code**

Tested on `cherow@1.6.8`.

```js
const cherow = require('cherow')

const code = `/*
Module documentation.
*/

/*
Documentation for variable \`point\`.
*/
const point = {
  /* This is a block comment inside an object expression */
  // This comment is for the x property
  x: 0,
  // …and this one is for the y property
  y: 1,
  // Trailing comment
}

class A {
  /*
	Documentation for class A.
	*/

  m(x, y = true) {
    /*
		Document for method \`m\` with params \`x\` and \`y\`.
		*/
  }
}
`

const ast = cherow.parse(code, {
  ecmaVersion: 10,
  module: true,
  loc: true,
  comments: true,
})

ast.comments.forEach(comment => {
  console.log(comment.loc)
  console.log(`\`${comment.value}\``)
  console.log('---')
})
```

**Expected behavior:**

```
{ start: { line: 1, column: 0 }, end: { line: 3, column: 2 } }
`
Module documentation.
`
---
{ start: { line: 5, column: 0 }, end: { line: 7, column: 2 } }
`
Documentation for variable `point`.
`
---
{ start: { line: 9, column: 2 }, end: { line: 9, column: 59 } }
` This is a block comment inside an object expression `
---
{ start: { line: 10, column: 2 }, end: { line: 10, column: 39 } }
` This comment is for the x property
`
---
{ start: { line: 12, column: 2 }, end: { line: 12, column: 40 } }
` …and this one is for the y property
`
---
{ start: { line: 14, column: 2 }, end: { line: 14, column: 21 } }
` Trailing comment
`
---
{ start: { line: 18, column: 2 }, end: { line: 20, column: 3 } }
`
	Documentation for class A.
	`
---
{ start: { line: 23, column: 4 },
  end: { line: 25, column: 4 } }
`
		Document for method `m` with params `x` and `y`.
		`
---
```

**Actual behavior:**

```
{ start: { line: 1, column: 0 }, end: { line: 1, column: 0 } }
`
Module documentation.
`
---
{ start: { line: 1, column: 0 }, end: { line: 1, column: 0 } }
`
Documentation for variable `point`.
`
---
{ start: { line: 9, column: 2 }, end: { line: 8, column: 15 } }
` This is a block comment inside an object expression `
---
{ start: { line: 10, column: 2 }, end: { line: 8, column: 15 } }
` This comment is for the x property
`
---
{ start: { line: 12, column: 2 }, end: { line: 11, column: 7 } }
` …and this one is for the y property
`
---
{ start: { line: 14, column: 2 }, end: { line: 13, column: 7 } }
` Trailing comment
`
---
{ start: { line: 18, column: 2 }, end: { line: 17, column: 9 } }
`
	Documentation for class A.
	`
---
{ start: { line: 23, column: 4 },
  end: { line: 22, column: 18 } }
`
		Document for method `m` with params `x` and `y`.
		`
---
```

**Comment by KFlash** [comment]
Thanx for the report. I may think this fixed I v.2.0? Latest version will be finished soon

**Comment by davidbonnet** [comment]
Cool! Haven't checked v2 yet but will soon.

**Comment by KFlash** [comment]
I'm closing this as it will be permanent fix I 2.0
